xcvm/cosmwasm/contracts/asset-registry/src/contract.rs

```diff
@@ -0,0 +1,168 @@
+use std::collections::BTreeMap;
+
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{to_binary, Binary, Deps, DepsMut, Env, MessageInfo, Response, StdResult};
+
+use crate::{
+	error::ContractError,
+	msg::{ExecuteMsg, GetAssetContractResponse, InstantiateMsg, QueryMsg},
+};
+
+use crate::state::{XcvmAssetId, ASSETS};
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+	_deps: DepsMut,
+	_env: Env,
+	_info: MessageInfo,
+	_msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+	Ok(Response::default())
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+	deps: DepsMut,
+	_env: Env,
+	_info: MessageInfo,
+	msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+	match msg {
+		ExecuteMsg::SetAssets(asset) => handle_set_assets(deps, asset),
+	}
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn query(deps: Deps, _env: Env, msg: QueryMsg) -> StdResult<Binary> {
+	match msg {
+		QueryMsg::GetAssetContract(token_id) => to_binary(&query_asset_contract(deps, token_id)?),
+	}
+}
+
+pub fn handle_set_assets(
```

Comment:
Might be good to have an add/remove asset too? Incremental changes

Reply:
We discussed this with @hussein-aitlahcen and decided on having only `set` function. What do you think @hussein-aitlahcen?

Reply:
Let's keep it like this and optimize it later